### DISCUSSION TITLE
Sentry SDK 2.0

### DISCUSSION THREAD

**Discussion by antonpirker** [getsentry/sentry-python#2980] [Announcements]
We will release a major update to the Sentry SDK, version 2.0.

### Major goal of the update:
- Refactor internal code and streamline public API to make it easier to create integrations and thus making it easier to support more libraries/frameworks out of the box. 
- Modernize code base and tooling to keep being future-proof.

### Language version support changes:
- Dropped support for Python 2.x.
- Dropped support for Python 3.5. 
- The SDK now requires at least Python 3.6!

### Framework version support changes:
- Dropped support for Celery 3.x.
- Dropped support for Django 1.8, 1.9, 1.10.
- Dropped support for Flask 0.x.
- Dropped support for gRPC < 1.39.
- Dropped support for Tornado < 6.

### Backwards compatibility
This update is mostly backwards compatible. If you do not do any custom instrumentation the update works out of the box. 
Have a look at our [migration guide](https://docs.sentry.io/platforms/python/migration/1.x-to-2.x) or the [changelog](https://github.com/getsentry/sentry-python/blob/sentry-sdk-2.0/CHANGELOG.md) to learn about how to upgrade when you do custom instrumentation. 

### Release Timeline
- Alpha releases: February 2024
- Release candidates: March and April 2024 (do some final polishing)
- Use release candidates as soon as possible on Sentry.io to [eat our own dog food](https://en.wikipedia.org/wiki/Eating_your_own_dog_food). 
- Final release of Sentry SDK 2.0: Mid April 2024

### Feedback
If you have any questions or feedback to the new release you can use this discussion, [Discord](https://discord.gg/Ww9hbqr) in the [#sdk-chat](https://discord.com/channels/621778831602221064/1211958154984820776) channel or create a [GitHub Issue](https://github.com/getsentry/sentry-python/issues)

**Comment by antonpirker** [comment]
Dogfooding ftw! We wanted to release today, but while dogfooding we found a regression in our Celery integration, so will fix this first and then make another release candidate.

**Comment by antonpirker** [comment]
In the `rc6` we released yesterday, the regression mentioned above is fixed. We will now use this rc on `sentry.io` itself for a couple of days to make sure everything works as expected and then I guess nothing is blocking the 2.0 release anymore!

**Comment by sentrivana** [comment]
[2.0](https://github.com/getsentry/sentry-python/releases/tag/2.0.0) is now out!

Already tried it out? Let us know how the migration went!

Didn't try it out yet? Let us know why -- this is also valuable feedback that'll help us make the migration path smoother.

**Comment by gregbrowndev** [comment]
Hi @sentrivana, how long will v1 SDK continue to be supported? Is it now only receiving security patches, or is it EOL?

I'm helping to support [a Sentry integration for Temporal users](https://github.com/temporalio/samples-python/pull/140) and want to determine whether we should keep our docs/example for the v1 SDK or recommend everyone move to v2. If v1 is EOL, then we could clean up Temporal's integration docs

Thanks 🙏🏻

**Reply by sentrivana** [reply]
Hey @gregbrowndev, thanks for reaching out. v1 is now only receiving security patches and we will likely EOL it with the release of v3 of the SDK, which is coming in the next few months.
